test/metabase/query_processor/middleware/metrics_test.clj

```diff
@@ -504,3 +504,51 @@
                                  {:source-query {:source-table (meta/id :checkins)
                                                  :joins        [{:condition    [:= [:field 1 nil] 2]
                                                                  :source-query before}]}}))))))))
+
+(deftest ^:parallel model-based-metric-use-test
+  (let [model {:lib/type :metadata/card
+               :id (fresh-card-id meta/metadata-provider)
+               :database-id (meta/id)
+               :name "Mock Model"
+               :type :model
+               :dataset-query (-> meta/metadata-provider
+                                  (lib/query (meta/table-metadata :products))
```

Comment:
A couple more tests I would like to see:

- A metric based on a model that only returns a custom expression field
- A metric based on a model that only returns an aggregate field

Are these allowed?

- A metric based on a model based on another model.
- A metric based on a multi-stage model.
- A metric based on a native query model.